client/signup/steps/choose-service/index.tsx

```diff
@@ -0,0 +1,143 @@
+import QueryProductsList from 'calypso/components/data/query-products-list';
+import { preventWidows } from 'calypso/lib/formatting';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import {
+	getProductDisplayCost,
+	isProductsListFetching,
+} from 'calypso/state/products-list/selectors';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { mouse, headset } from '../../icons';
+import { ChoiceType } from './types';
+import type { SelectItem } from '@automattic/onboarding';
+
+import './style.scss';
+
+interface Props {
+	existingSiteCount: number;
+	goToNextStep: () => void;
+	submitSignupStep: ( { stepName, wasSkipped }: { stepName: string; wasSkipped: boolean } ) => void;
+	goToStep: ( stepName: string ) => void;
+	stepName: string;
+	queryObject: {
+		siteSlug?: string;
+		siteId?: string;
+	};
+}
+
+type DIFMOrBuiltByIntent = SelectItem< ChoiceType >;
+
+const Placeholder = () => <span className="choose-service__placeholder">&nbsp;</span>;
+
+export default function ChooseServiceStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const displayCost = useSelector( ( state ) => getProductDisplayCost( state, WPCOM_DIFM_LITE ) );
+	const isLoading = useSelector( isProductsListFetching );
+
+	const headerText = translate( 'Let our experts create your dream site' );
+
+	const subHeaderText = translate(
+		'Get your business up quickly with our experts building your responsive, professional website. Select your best option.'
+	);
+
+	const intents: DIFMOrBuiltByIntent[] = [
+		{
+			key: 'difm',
+			title: translate( 'Do It For Me' ),
+			description: translate(
+				'Get a professionally designed, mobile-optimized website in %(fulfillmentDays)d business days or less for a one-time fee of {{PriceWrapper}}%(displayCost)s{{/PriceWrapper}} plus a one year subscription of the %(plan)s plan.',
+				{
+					args: {
+						displayCost,
+						fulfillmentDays: 4,
+						plan: isEnabled( 'plans/pro-plan' )
+							? getPlan( PLAN_WPCOM_PRO )?.getTitle()
+							: getPlan( PLAN_PREMIUM )?.getTitle(),
+					},
+					components: {
+						PriceWrapper: isLoading ? <Placeholder /> : <strong />,
+					},
+				}
+			),
+			icon: mouse,
+			value: 'difm',
+			actionText: translate( 'Get Started' ),
+			isPrimary: true,
+		},
+		{
+			key: 'builtby',
+			title: translate( 'Built by WordPress.com - Concierge' ),
+			description: translate(
+				'Perfect for those looking for advanced features (e.g., e-commerce, migrations), customized designs, or who prefer a custom, consultative experience.'
+			),
+			icon: headset,
+			value: 'builtby',
+			actionText: translate( 'Learn More' ),
+		},
+	];
+
+	useEffect( () => {
+		if ( ! props.queryObject.siteSlug && ! props.queryObject.siteId ) {
+			throw new Error(
+				'website-design-services did not provide the site slug or site id it is configured to.'
+			);
+		}
```

Comment:
NIT. and not something to address I think! I wish there's a better / more friendly way to handle exceptions like these. This is obviously a fatal error and not something to expect to happen under normal right? but the blank screen and the console error are not very friendly or informative I think. 

What do you think? 
(I'm guessing is there was some Error boundary to catch and display something to the user may be better?)

Reply:
I agree, this is not the a good experience when this happens. I followed the example from here:
https://github.com/Automattic/wp-calypso/blob/944d1e7a185e7a454ed92bf9e6a38b2941c0bc68/client/landing/stepper/declarative-flow/site-setup-flow.ts#L354-L356

This situation can only happen if the URL is edited manually. IMO, we should then redirect to a 404/error page. Can we do this in a follow-up PR?